apps/ui/src/stores/votingPowers.ts

```diff
@@ -0,0 +1,99 @@
+import { utils } from '@snapshot-labs/sx';
+import { defineStore } from 'pinia';
+import { getNetwork } from '@/networks';
+import { VotingPower, VotingPowerStatus } from '@/networks/types';
+import { Proposal, Space } from '@/types';
+
+const LATEST_BLOCK_NAME = 'latest';
+
+type VotingPowerItem = {
+  votingPowers: VotingPower[];
+  totalVotingPower: bigint;
+  status: VotingPowerStatus;
+  symbol: string;
+  decimals: number;
+  error: utils.errors.VotingPowerDetailsError | null;
+};
+
+export const useVotingPowersStore = defineStore('votingPowers', () => {
+  const votingPowers = reactive<Map<string, VotingPowerItem>>(new Map());
```

Comment:
Not sure why it's reactive. Right now we lose reactivity as we access it through `get` method and it loses its reactivity there.

We need to access it directly in `computed` if we want to make it truly reactive:
```diff
diff --git a/apps/ui/src/composables/useVotingPower.ts b/apps/ui/src/composables/useVotingPower.ts
index 6371081f..35098a8e 100644
--- a/apps/ui/src/composables/useVotingPower.ts
+++ b/apps/ui/src/composables/useVotingPower.ts
@@ -1,4 +1,5 @@
 import { supportsNullCurrent } from '@/networks';
+import { getIndex as getVotingPowerIndex } from '@/stores/votingPowers';
 import { Proposal, Space } from '@/types';
 
 export function useVotingPower() {
@@ -28,7 +29,11 @@ export function useVotingPower() {
   });
 
   const votingPower = computed(
-    () => space.value && votingPowersStore.get(space.value, block.value)
+    () =>
+      space.value &&
+      votingPowersStore.votingPowers.get(
+        getVotingPowerIndex(space.value, block.value)
+      )
   );
 
   const hasVoteVp = computed(
diff --git a/apps/ui/src/stores/votingPowers.ts b/apps/ui/src/stores/votingPowers.ts
index b036f1bc..a22b212b 100644
--- a/apps/ui/src/stores/votingPowers.ts
+++ b/apps/ui/src/stores/votingPowers.ts
@@ -15,20 +15,13 @@ type VotingPowerItem = {
   error: utils.errors.VotingPowerDetailsError | null;
 };
 
+export function getIndex(space: Space, block: number | null): string {
+  return `${space.id}:${block ?? LATEST_BLOCK_NAME}`;
+}
+
 export const useVotingPowersStore = defineStore('votingPowers', () => {
   const votingPowers = reactive<Map<string, VotingPowerItem>>(new Map());
 
-  function getIndex(space: Space, block: number | null): string {
-    return `${space.id}:${block ?? LATEST_BLOCK_NAME}`;
-  }
-
-  function get(
-    space: Space,
-    block: number | null
-  ): VotingPowerItem | undefined {
-    return votingPowers.get(getIndex(space, block));
-  }
-
   async function fetch(
     item: Space | Proposal,
     account: string,
@@ -36,7 +29,7 @@ export const useVotingPowersStore = defineStore('votingPowers', () => {
   ) {
     const space: Space = 'space' in item ? (item.space as Space) : item;
 
-    const existingVotingPower = get(space, block);
+    const existingVotingPower = votingPowers.get(getIndex(space, block));
     if (existingVotingPower && existingVotingPower.status === 'success') return;
 
     const network = getNetwork(item.network);
@@ -92,7 +85,7 @@ export const useVotingPowersStore = defineStore('votingPowers', () => {
   }
 
   return {
-    get,
+    votingPowers,
     fetch,
     reset
   };
```